packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -570,7 +571,7 @@ export function resetShadowRoot(vm: VM) {
 }
 
 export function scheduleRehydration(vm: VM) {
-    if (!vm.isScheduled) {
+    if (isFalse(vm.isScheduled) && isFalse(vm.renderer.ssr)) {
```

Comment:
Nit: For me, it's more clear what's going on when you have something like the following at the top of the function body:
```
if (isTrue(vm.renderer.ssr)) {
    return;
}
```